### PR TITLE
Docker fixes and parameter cleanups

### LINKED ISSUE #28: Trajectory resolution should be a timestep in seconds, not number of points
Right now, the `trajectory_res` parameter in the KTOpt planner defaults to 101 points for the trajectory, but actually it should ve a time resolution (e.g. 0.01 seconds) and then dependent on the trajectory length instead.

### PR DESCRIPTION
**Pull request by sea-bass**
As the title says, was just going through and found some slight fixes to the Docker workflow... and during rebuilds I fixed a few other small things as well. 

The important stuff is:

* The `TERM=xterm-256color` env variable had a typo in it, which gave errors when trying to run bash commands in the container like `clear`
* There was one more extra thing needed in the entrypoint for `pre-commit run -a` to work correctly.
* Changed some collision check constraints from constants to actual parameters.
* Changed the trajectory time resolution to seconds (Closes #28)

### CODE REVIEW

**Comment by sea-bass** [comment]
@kamiradi FYI (can't officially put you as a reviewer)

File: res/ktopt_moveit_parameters.yaml [hunk 1]
```diff
@@ -1,19 +1,41 @@
 ktopt_interface:
   num_iterations: {
     type: int,
-    description: "dummy",
+    description: "Number of iterations for the Drake mathematical program solver.",
     default_value: 1000,
     validation: {
       gt_eq<>: [1]
     }
```
Comment (sea-bass) [inline res/ktopt_moveit_parameters.yaml:8]: Actually, I realize this parameter is not currently being used, as we're using default options in the mathematical program `Solve()` function.

**Comment by sea-bass** [comment]
@sjahr I am seeing a segfault now, which is likely due to some recent changes in `moveit2` repo... it's in a function that was added in the planning scene's copy constructor in https://github.com/moveit/moveit2/pull/2858 -- so I think there is an issue here.

```
[pipeline_testbench_example-4] Stack trace (most recent call last):
[pipeline_testbench_example-4] #10   Object "", at 0xffffffffffffffff, in 
[pipeline_testbench_example-4] #9    Object "/root/workspace/install/moveit_drake/lib/moveit_drake/pipeline_testbench_example", at 0x5c054e675b64, in _start
[pipeline_testbench_example-4] #8    Object "/usr/lib/x86_64-linux-gnu/libc.so.6", at 0x7b37c57e7e3f, in __libc_start_main
[pipeline_testbench_example-4] #7    Object "/usr/lib/x86_64-linux-gnu/libc.so.6", at 0x7b37c57e7d8f, in 
[pipeline_testbench_example-4] #6    Object "/root/workspace/install/moveit_drake/lib/moveit_drake/pipeline_testbench_example", at 0x5c054e6756c3, in main
[pipeline_testbench_example-4] #5    Object "/root/workspace/install/moveit_drake/lib/moveit_drake/pipeline_testbench_example", at 0x5c054e682513, in pipeline_testbench::Demo::planAndVisualize(std::vector<pipeline_testbench::Demo::PipelineConfig, std::allocator<pipeline_testbench::Demo::PipelineConfig> >, moveit_msgs::msg::MotionPlanRequest_<std::allocator<void> > const&)
[pipeline_testbench_example-4] #4    Object "/root/workspace/install/moveit_core/lib/libmoveit_planning_scene.so.2.10.0", at 0x7b37c5f5a339, in planning_scene::PlanningScene::clone(std::shared_ptr<planning_scene::PlanningScene const> const&)
[pipeline_testbench_example-4] #3    Object "/root/workspace/install/moveit_core/lib/libmoveit_planning_scene.so.2.10.0", at 0x7b37c5f4ea15, in planning_scene::PlanningScene::diff() const
[pipeline_testbench_example-4] #2    Object "/root/workspace/install/moveit_core/lib/libmoveit_planning_scene.so.2.10.0", at 0x7b37c5f4e77b, in planning_scene::PlanningScene::PlanningScene(std::shared_ptr<planning_scene::PlanningScene const> const&)
[pipeline_testbench_example-4] #1    Object "/root/workspace/install/moveit_core/lib/libmoveit_planning_scene.so.2.10.0", at 0x7b37c5f49568, in planning_scene::PlanningScene::setCollisionObjectUpdateCallback(std::function<void (std::shared_ptr<collision_detection::World::Object const> const&, collision_detection::World::Action)> const&)
[pipeline_testbench_example-4] #0    Object "/root/workspace/install/moveit_core/lib/libmoveit_collision_detection.so.2.10.0", at 0x7b37c56fff1b, in collision_detection::World::addObserver(std::function<void (std::shared_ptr<collision_detection::World::Object const> const&, collision_detection::World::Action)> const&)
[pipeline_testbench_example-4] Segmentation fault (Address not mapped to object [0x40])
[ERROR] [pipeline_testbench_example-4]: process has died [pid 1495, exit code -11, cmd '/root/workspace/install/moveit_drake/lib/moveit_drake/pipeline_testbench_example --ros-args -r __node:=pipeline_testbench_example --params-file /tmp/launch_params_lurebq_7 --params-file /tmp/launch_params_ogckwz85 --params-file /tmp/launch_params_kjgpz5jw'].
```

**Comment by kamiradi** [comment]
> @sjahr I am seeing a segfault now, which is likely due to some recent changes in `moveit2` repo... it's in a function that was added in the planning scene in [moveit/moveit2#2858](https://github.com/moveit/moveit2/pull/2858) -- so I think there is an issue here.
> 
> EDIT: Yep, I reverted that change and this fixes the stack trace.
> 
> ```
> [pipeline_testbench_example-4] Stack trace (most recent call last):
> [pipeline_testbench_example-4] #10   Object "", at 0xffffffffffffffff, in 
> [pipeline_testbench_example-4] #9    Object "/root/workspace/install/moveit_drake/lib/moveit_drake/pipeline_testbench_example", at 0x5c054e675b64, in _start
> [pipeline_testbench_example-4] #8    Object "/usr/lib/x86_64-linux-gnu/libc.so.6", at 0x7b37c57e7e3f, in __libc_start_main
> [pipeline_testbench_example-4] #7    Object "/usr/lib/x86_64-linux-gnu/libc.so.6", at 0x7b37c57e7d8f, in 
> [pipeline_testbench_example-4] #6    Object "/root/workspace/install/moveit_drake/lib/moveit_drake/pipeline_testbench_example", at 0x5c054e6756c3, in main
> [pipeline_testbench_example-4] #5    Object "/root/workspace/install/moveit_drake/lib/moveit_drake/pipeline_testbench_example", at 0x5c054e682513, in pipeline_testbench::Demo::planAndVisualize(std::vector<pipeline_testbench::Demo::PipelineConfig, std::allocator<pipeline_testbench::Demo::PipelineConfig> >, moveit_msgs::msg::MotionPlanRequest_<std::allocator<void> > const&)
> [pipeline_testbench_example-4] #4    Object "/root/workspace/install/moveit_core/lib/libmoveit_planning_scene.so.2.10.0", at 0x7b37c5f5a339, in planning_scene::PlanningScene::clone(std::shared_ptr<planning_scene::PlanningScene const> const&)
> [pipeline_testbench_example-4] #3    Object "/root/workspace/install/moveit_core/lib/libmoveit_planning_scene.so.2.10.0", at 0x7b37c5f4ea15, in planning_scene::PlanningScene::diff() const
> [pipeline_testbench_example-4] #2    Object "/root/workspace/install/moveit_core/lib/libmoveit_planning_scene.so.2.10.0", at 0x7b37c5f4e77b, in planning_scene::PlanningScene::PlanningScene(std::shared_ptr<planning_scene::PlanningScene const> const&)
> [pipeline_testbench_example-4] #1    Object "/root/workspace/install/moveit_core/lib/libmoveit_planning_scene.so.2.10.0", at 0x7b37c5f49568, in planning_scene::PlanningScene::setCollisionObjectUpdateCallback(std::function<void (std::shared_ptr<collision_detection::World::Object const> const&, collision_detection::World::Action)> const&)
> [pipeline_testbench_example-4] #0    Object "/root/workspace/install/moveit_core/lib/libmoveit_collision_detection.so.2.10.0", at 0x7b37c56fff1b, in collision_detection::World::addObserver(std::function<void (std::shared_ptr<collision_detection::World::Object const> const&, collision_detection::World::Action)> const&)
> [pipeline_testbench_example-4] Segmentation fault (Address not mapped to object [0x40])
> [ERROR] [pipeline_testbench_example-4]: process has died [pid 1495, exit code -11, cmd '/root/workspace/install/moveit_drake/lib/moveit_drake/pipeline_testbench_example --ros-args -r __node:=pipeline_testbench_example --params-file /tmp/launch_params_lurebq_7 --params-file /tmp/launch_params_ogckwz85 --params-file /tmp/launch_params_kjgpz5jw'].
> ```
This segfault relates to the Planning Scene not being set before the planning scene transcription begins. Spent some time debugging this in the last PR

**Review by sjahr — APPROVED** [review]
I'll take a look into the moveit issue but these changes look good!

**Comment by kamiradi** [comment]
@sea-bass What change did you revert this to. I was unable to get rid of the segfault.

**Comment by sjahr** [comment]
@kamiradi Check out https://github.com/moveit/moveit2/pull/2950

**Pull request merged**